acceptancetests/assess_lxd_space.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+"""Testing spaces and subnets settings for app deployment to lxd container.
+   This is an validation of Bug 1685782:
+   https://bugs.launchpad.net/juju/+bug/1685782
+   The test procedures are based on comment #29.
+"""
+
+from __future__ import print_function
+
+import os
+import sys
+import json
+import logging
+import argparse
+
+from jujucharm import local_charm_path
+from deploy_stack import BootstrapManager
+
+from utility import (
+    JujuAssertionError,
+    add_basic_testing_arguments,
+    configure_logging,
+    )
+
+__metaclass__ = type
+log = logging.getLogger("assess_lxd_space")
+
+
+def assert_initial_spaces(client):
+    """Initial spaces status after bootstrap should show as:
+       no spaces to display"""
+    # Run juju spaces --format json
+    # Bug 1704105: https://bugs.launchpad.net/juju/+bug/1704105
+    # merge_stderr need to be set to True for the initial assert
+    # because spaces list is empty at this point.
+    raw_output = client.get_juju_output(
+        'spaces', '--format', 'json', include_e=False, merge_stderr=True)
+    expected_spaces = 'no spaces to display'
+    if expected_spaces not in raw_output:
+        raise JujuAssertionError(
+            'Incorrect initial spaces status. Found: {}\nExpected: {}'.format(
+            raw_output, expected_spaces))
+    else:
+        log.info('Initial spaces status is: {}.'.format(expected_spaces))
+
+
+def get_subnets(client):
+    # Run juju subnets --format json
+    raw_output = client.get_juju_output(
+        'subnets', '--format', 'json', include_e=False)
+    try:
+        subnets_output = json.loads(raw_output)
+    except ValueError as e:
+        log.error('Invalid output from juju subnets:\n{}'.format(raw_output))
```

Comment:
The message in log.error() here is human language to tell user that the output from juju subnets is wrong. In comparison the traceback from exception are much harder to understand - it may go with multiple lines, python's native function name and line number, etc.
Usually, if a thing we check doesn't exist, the corresponding Juju command won't give an valid json object if using --format json, e.g.: if there is no space exists then juju space --format json won't give a valid json for parsing. However, I didn't try juju subnets under the no subnet exists condition, just give it a same check as juju space.
Not necessarily, as explained above.